src/js/components/DataSearch/DataSearch.js

```diff
@@ -42,17 +67,14 @@ export const DataSearch = ({ drop, id: idProp, responsive, ...rest }) => {
       name="_search"
       icon={<Search />}
       type="search"
+      onChange={onChange}
```

Comment:
This is looking good for `onChange` approach. Do you envision there would be a way to incorporate the `updateOn="submit"` functionality with this direction? https://github.com/grommet/grommet/pull/7029

Reply:
Yes. I think it just needs a Keyboard onEnter handler since it can't rely on the form submit anymore

Reply:
I do have updateOn tweaks stashed. I can either add them to this PR or wait until this merges and make them a commit to the updateOn PR since it will have to be updated or closed anyway.

Reply:
Let's handle it on the updateOn PR for clarity.